gbpservice/neutron/nsf/configurator/agents/firewall.py

```diff
@@ -0,0 +1,323 @@
+
+    def handle_event(self, ev):
+        try:
+            msg = ("Worker process with ID: %s starting to "
+                   "handle task: %s of type firewall. "
+                   % (os.getpid(), ev.id))
+            LOG.debug(msg)
+
+            driver = self._get_driver()
+            self.method = getattr(driver, "%s" % (ev.id.lower()))
+            self.invoke_driver_for_plugin_api(ev)
+        except Exception as err:
+            LOG.error("Failed to perform the operation: %s. %s"
+                      % (ev.id, str(err).capitalize()))
+
+    def invoke_driver_for_plugin_api(self, ev):
+        context = ev.data.get('context')
+        firewall = ev.data.get('firewall')
+        host = ev.data.get('host')
+
+        if ev.id == 'CREATE_FIREWALL':
+            if not self._is_firewall_rule_exists(firewall):
+                msg = ("Firewall status set to ACTIVE")
+                LOG.debug(msg)
+                return self.plugin_rpc.set_firewall_status(
+                    context, firewall['id'], const.STATUS_ACTIVE)
+            # Added to handle in service vm agents. VM agent will add
+            # default DROP rule.
+            # if not self._is_firewall_rule_exists(firewall):
+            #     self.plugin_rpc.set_firewall_status(
+            #         context, firewall['id'], const.STATUS_ACTIVE)
+            try:
+                status = self.method(context, firewall, host)
+            except Exception as err:
+                self.plugin_rpc.set_firewall_status(
+                    context, firewall['id'], const.STATUS_ERROR)
+                msg = ("Failed to configure Firewall and status is "
+                       "changed to ERROR. %s." % str(err).capitalize())
+                LOG.error(msg)
+            else:
+                self.plugin_rpc.set_firewall_status(
+                    context, firewall['id'], status)
+                msg = ("Configured Firewall and status set to %s" % status)
+                LOG.info(msg)
+
+        elif ev.id == 'DELETE_FIREWALL':
+            if not self._is_firewall_rule_exists(firewall):
+                return self.plugin_rpc.firewall_deleted(context,
+                                                        firewall['id'])
+            try:
+                status = self.method(context, firewall, host)
+            except ConnectionError:
+                # FIXME(Vikash) It can't be correct everytime
+                LOG.warning("There is a connection error for firewall %r of "
+                            "tenant %r. Assuming either there is serious "
+                            "issue with VM or data path is completely "
+                            "broken. For now marking that as delete."
+                            % (firewall['id'], firewall['tenant_id']))
+                self.plugin_rpc.firewall_deleted(context, firewall['id'])
+
+            except Exception as err:
+                # TODO(Vikash) Is it correct to raise ? As the subsequent
```

Comment:
Lets not have names in code. Remove them. Lets understand what issue is pointed out here.